src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.cs

```diff
@@ -32,11 +40,13 @@ public SafeSocketHandle(IntPtr preexistingHandle, bool ownsHandle)
 
         private SafeSocketHandle() : base(true) { }
 
-        private InnerSafeCloseSocket _innerSocket;
+        private bool TryOwnClose()
+        {
+            return handle != new IntPtr(-1) &&
```

Comment:
What's the purpose of the handle != -1 check?  I'm wondering if it'd be safer to just always do the CompareExchange.

Reply:
This happens for invalid handles. I'll refactor this.